pkg/reloader/reloader.go

```diff
@@ -167,6 +175,27 @@ func (r *Reloader) Watch(ctx context.Context) error {
 			if _, ok := watchables[filepath.Dir(event.Name)]; !ok {
 				continue
 			}
+
+			// When the file's inode changed, watcher will received an 'op=CHMOD' event.
+			// Watcher will never receive any events again because the original file inode is invalid.
+			// This will happen when you using 'mv' command to overwrite config file.
+			// Also, when secret updated on kubernetes. It will generate new files with new inode.
+			// Config file should restart watching in this case.
+			if event.Op == fsnotify.Chmod {
+				if inode, err := getFileInode(r.cfgFile); err != nil {
+					return err
+				} else if r.lastCfgInode == inode {
+					continue
+				} else {
+					if err := watcher.Add(r.cfgFile); err != nil {
```

Comment:
Doesn't this leak file descriptors if we do not remove the old one?

Reply:
fsnotify will remove the file watching when it finds that the original file has disappeared. So i did't remove it here. Just simply add it back.